packages/cozy-harvest-lib/src/components/AccountEditor.jsx

```diff
@@ -0,0 +1,45 @@
+import React, { PureComponent } from 'react'
+import PropTypes from 'react-proptypes'
+
+import { withMutations } from 'cozy-client'
+
+import AccountForm from './AccountForm'
+import { accountsMutations } from '../connections/accounts'
+import { hydrateAccount } from '../helpers/accounts'
+
+/**
+ * Encapsulates an AccocuntForm of an existing account, and allow to update it.
+ */
+export class AccountEditor extends PureComponent {
+  update = async data => {
+    const { account, onUpdate, onUpdateSuccess, updateAccount } = this.props
+    onUpdate(account)
+    const updatedAccount = await updateAccount(hydrateAccount(account, data))
+    onUpdateSuccess(updatedAccount)
+  }
+
+  render() {
+    const { account, konnector, submitting } = this.props
+    return (
+      <AccountForm
+        fields={konnector.fields}
+        initialValues={account && (account.auth || account.oauth)}
+        locales={konnector.locales}
+        oauth={konnector.oauth}
+        onSubmit={this.update}
+        submitting={submitting}
+      />
+    )
+  }
+}
+
+AccountEditor.propTypes = {
+  account: PropTypes.object.isRequired,
```

Comment:
I like to describe the shape of the object we want

Reply:
Do you have an example somewhere ?

Reply:
Yep : 
https://github.com/cozy/cozy-contacts/blob/c42e5deda439b91705e28cd2ca77bff5d8400519/src/components/ContactPropTypes.js#L3 
https://github.com/cozy/cozy-contacts/blob/3336b6d9d23771ab1533aebafabd358e7af60629/src/components/Modals/ContactCardMenu.jsx#L31
https://github.com/cozy/cozy-drive/blob/044ac1db3b9f9cc6738547f132f81bf2d7989a96/src/drive/web/modules/drive/FileOpenerExternal.jsx#L72

Reply:
Thanks.

It's look like a job for another day, what do you think. IMHO this kind of PropTypes definition should be mutualized in cozy-doctypes and imported into apps and libs. What do you think ?

Reply:
yep